reportek/core/models/reporting.py

```diff
@@ -213,35 +213,62 @@ def validate_filename(value):
         )
 
 
-class EnvelopeOriginalFile(models.Model):
+class EnvelopeFileQuerySet(models.QuerySet):
+
+    def delete(self):
+        for obj in self:
+            try:
+                os.remove(obj.file.path)
+                debug(f'Deleted disk file {obj.file.path}')
+            except FileNotFoundError:
+                error(f'Could not delete envelope file from disk (not found): '
+                      f'{obj.file.path}')
+        super().delete()
+
+
+class EnvelopeFileCommon(models.Model):
     """
-    Used to describe original non-XML files uploaded to the envelope
-    that are converted to and stored as XML EnvelopeFiles.
+    Abstract base class for EnvelopeFile and EnvelopeOriginalFile,
+    holding some common fields for these two.
     """
 
     class Meta:
-        db_table = 'original_envelope_file'
+        abstract = True
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        # keep track of the name to handle renames
+        self._prev_name = (None if self.pk is None
+                           else self.name)
 
     def get_envelope_directory(self, filename):
         return os.path.join(
             self.envelope.get_storage_directory(),
             os.path.basename(filename)
         )
 
-    envelope = models.ForeignKey(Envelope, related_name='original_files')
+    # Used by get_download_url()
+    _download_view_name = ''
+
+    envelope = models.ForeignKey(Envelope, related_name='%(class)ss')
```

Comment:
There might be a way to avoid lots of redundant typing elsewhere (`envelope.envelopefiles`) caused by the dynamic class-derived `related_name`, by replacing the string template with a function that strips the 'envelope' prefix from the class name, if present (see https://stackoverflow.com/a/43093779)